### PR TITLE
add message length check logic

### PR DESCRIPTION
**Pull request by xxo1shine**

### CODE REVIEW

File: src/main/java/org/tron/p2p/utils/ProtoUtil.java [hunk 1]
```diff
@@ -23,11 +25,21 @@ public static Connect.CompressMessage compressMessage(byte[] data) throws IOExce
...
     if (message.getType().equals(Connect.CompressMessage.CompressType.uncompress)) {
-      return message.getData().toByteArray();
-    } else {
-      return Snappy.uncompress(message.getData().toByteArray());
+      return data;
+    }
+
+    int length = Snappy.uncompressedLength(data);
+    if (length >= Parameter.MAX_MESSAGE_LENGTH) {
+      throw new IOException("message is too big, len=" + length);
```
Comment (jwrct) [inline src/main/java/org/tron/p2p/utils/ProtoUtil.java:35]: Suggest optimizing the exception information, it should also be that the message after decompression is too large.

Comment (317787106) [reply src/main/java/org/tron/p2p/utils/ProtoUtil.java:35]: These two IOException can be merged into one.

Comment (317787106) [reply src/main/java/org/tron/p2p/utils/ProtoUtil.java:35]: you can use P2pException MESSAGE_WITH_WRONG_LENGTH

Comment (ss3344520) [reply src/main/java/org/tron/p2p/utils/ProtoUtil.java]: Modified, the two checks cannot be combined, the length must be checked before decompression.

File: src/main/java/org/tron/p2p/utils/ProtoUtil.java [hunk 2]
```diff
@@ -22,12 +25,25 @@ public static Connect.CompressMessage compressMessage(byte[] data) throws IOExce
...
+          "message is too big, len=" + length);
+    }
+
+    byte[] d2 = Snappy.uncompress(data);
+    if (d2.length >= Parameter.MAX_MESSAGE_LENGTH) {
+      throw new P2pException(P2pException.TypeEnum.BIG_MESSAGE,
+          "uncompressed is too big, len=" + length);
     }
+    return d2;
   }
```
Comment (317787106) [inline src/main/java/org/tron/p2p/utils/ProtoUtil.java:47]: int length = Snappy.uncompressedLength(data);
byte[] data = null;
if(length < Parameter.MAX_MESSAGE_LENGTH){
 data = Snappy.uncompress(data);
}
if(length >= MAX_MESSAGE_LENGTH || d2.length >= Parameter.MAX_MESSAGE_LENGTH){
throws xx;
}

**Pull request merged**